### PR TITLE
THOR-480 Current Filters Rework

### PR DESCRIPTION
**Pull request by arciisine**

### CODE REVIEW

**Comment by arciisine** [comment]
@ThomasSchellenbergNextCentury So I'm getting a weird error in FilterService but that should be wholly unrelated, hopefully a second set of eyes will help

**Comment by arciisine** [comment]
> @ThomasSchellenbergNextCentury So I'm getting a weird error in FilterService but that should be wholly unrelated, hopefully a second set of eyes will help

Actually, figured it out, everything should be good now.

**Comment by ThomasSchellenbergNextCentury** [comment]
@arciisine Please run `npm run-script lint` and fix the lint errors.

File: src/app/components/current-filters/current-filters.component.ts [hunk 1]
```diff
@@ -13,30 +13,132 @@
...
+    value?: any;
+}
+
+interface FilterGroup {
+    name?: string;
+    multi?: boolean;
+    filters?: FilterDisplay[];
+}
+
+export class FilterUtil {
```
Comment (ThomasSchellenbergNextCentury) [inline src/app/components/current-filters/current-filters.component.ts]: @arciisine There's already an exported namespace called FilterUtil (in the FilterService) so we should probably rename one of them.

File: src/app/services/filter.service.ts [hunk 2]
```diff
@@ -628,12 +628,22 @@ export class FilterService {
...
+        return this.getRawFilters(filterDataSourceList).map(filter => filter.toDesign());
+    }
+
+    /**
+     * Returns the raw filters for the given data sources, or all filters if no data sources are given.
+     *
+     * @arg {FilterDataSource[]} [filterDataSourceList]
+     * @return {AbstractFilter[]}
+     */
+    public getRawFilters(filterDataSourceList?: FilterDataSource[]) {
```
Comment (ThomasSchellenbergNextCentury) [inline src/app/services/filter.service.ts]: @arciisine Please add a return type to the function definition.

File: src/app/components/current-filters/current-filters.component.ts [hunk 3]
```diff
@@ -13,30 +13,132 @@
...
+            case 'date': return op.replace(/^<=?$/, 'before').replace(/^>=?$/, 'after');
+            default: return op.replace(/^=$/g, '');
+        }
+    }
+
+    static isGeo(type: string, name: string) {
+        return type === 'double' && /[.](lat|lon)$/i.test(name);
+    }
+
+    static computeFilter(x: AbstractFilter): FilterDisplay {
```
Comment (ThomasSchellenbergNextCentury) [inline src/app/components/current-filters/current-filters.component.ts]: @arciisine Could we use a more descriptive variable name than `x`?

File: src/app/components/current-filters/current-filters.component.spec.ts [hunk 4]
```diff
@@ -14,21 +14,61 @@
...
 import { initializeTestBed } from '../../../testUtils/initializeTestBed';
 
 import { CurrentFiltersModule } from './current-filters.module';
 import { ConfigService } from '../../services/config.service';
+import { CompoundFilterType } from '../../services/abstract.search.service';
+import { SearchService } from '../../services/search.service';
+import moment = require('moment');
+
 
 describe('Component: CurrentFiltersComponent', () => {
```
Comment (ThomasSchellenbergNextCentury) [inline src/app/components/current-filters/current-filters.component.spec.ts:33]: @arciisine I'd recommend adding a few more tests on computeFilter:
- Compound filters that aren't date ranges or bounding boxes
- Non-date/non-geo filters that use `>=` or `<=` (verify that translateOperator is called on non-date/non-geo filters)
- Compound filters with multiple levels of nesting

**Comment by ThomasSchellenbergNextCentury** [comment]
FYI I create a ticket to move the behavior from computeFilters into the Filter classes based on our discussion today:  https://nextcentury.atlassian.net/browse/THOR-1236

File: src/app/components/current-filters/current-filters.component.spec.ts (hunk 4 above)
Comment (arciisine) [reply src/app/components/current-filters/current-filters.component.spec.ts:33]: Pushed a test that hits both of these, should hopefully be good to go

File: src/app/components/current-filters/current-filters.component.spec.ts [hunk 5]
```diff
@@ -13,22 +13,62 @@
...
 
 import { initializeTestBed } from '../../../testUtils/initializeTestBed';
 
 import { CurrentFiltersModule } from './current-filters.module';
 import { ConfigService } from '../../services/config.service';
+import { CompoundFilterType } from '../../services/abstract.search.service';
+import { SearchService } from '../../services/search.service';
 
-describe('Component: CurrentFiltersComponent', () => {
+fdescribe('Component: CurrentFiltersComponent', () => {
```
Comment (ThomasSchellenbergNextCentury) [inline src/app/components/current-filters/current-filters.component.spec.ts]: @arciisine Whoops, there's an `fdescribe` in here.  Looks like I forgot to add that rule to the new linter!

Comment (arciisine) [reply src/app/components/current-filters/current-filters.component.spec.ts]: Yep, whoops.  Pushing out

**Pull request merged**